cli/tests/test_diff_command.rs

```diff
@@ -290,6 +290,75 @@ fn test_diff_types() {
     }
 }
 
+#[test]
+fn test_diff_name_only() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "repo"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    let file_path = repo_path.join("foo");
+
+    // Missing
+    test_env.jj_cmd_ok(&repo_path, &["new", "root()", "-m=missing"]);
+
+    // Normal file
+    test_env.jj_cmd_ok(&repo_path, &["new", "root()", "-m=file"]);
+    std::fs::write(&file_path, "foo").unwrap();
+
+    // Conflict (add/add)
+    test_env.jj_cmd_ok(&repo_path, &["new", "root()", "-m=conflict"]);
+    std::fs::write(&file_path, "foo").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["new", "root()"]);
+    std::fs::write(&file_path, "bar").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["squash", r#"--into=description("conflict")"#]);
+
+    #[cfg(unix)]
+    {
+        use std::os::unix::fs::PermissionsExt;
+        use std::path::PathBuf;
+
+        // Executable
+        test_env.jj_cmd_ok(&repo_path, &["new", "root()", "-m=executable"]);
+        std::fs::write(&file_path, "foo").unwrap();
+        std::fs::set_permissions(&file_path, std::fs::Permissions::from_mode(0o755)).unwrap();
+
+        // Symlink
+        test_env.jj_cmd_ok(&repo_path, &["new", "root()", "-m=symlink"]);
+        std::os::unix::fs::symlink(PathBuf::from("."), &file_path).unwrap();
+    }
```

Comment:
Since this diff mode doesn't care about the the file type, I'd rather copy from the `diff --summary` test. Oh, there isn't one... Maybe just include one addition, one modification, and one deletion and show that they alll show up then? Should we also put the files in a subdirectory to show that it's the full (but relative) path that's printed?